### PR TITLE
Fix another runtime

### LINKED ISSUE #9597: Joining the round as it starts occasionally FUCKS YOUR PROFILE AND LOADS A BLANK ONE
#### Description of issue

http://i.imgur.com/Vg5518v.png  This has occurred to a handful of players, including myself just now.  When you join a round before the server has finished loading it has a chance of this happening
#### Difference between expected and actual behavior

My character slot to be loaded or at least not get a blanked out name and shit

I get a blanked out name and no preferences saved both in game and in my settings
#### Steps to reproduce

???
#### Specific information for locating

Reload slot fixes it but you don't have any indication it has happened until you open Setup Preferences or join with a blank name.
#### Length of time in which bug has been known to occur

Since yall started meddling a month ago with profiles
#### Server revision

c1d94cf88087938c9d38b61d36edae2582d80d08 Current Infomational Settings: 
Protect Authority Roles From Tratior: 1
#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

### PR DESCRIPTION
**Pull request by clusterfack**
Intigracy fucked up
I believe this definitively fixes #9597

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -182,25 +182,27 @@ var/const/MAX_SAVE_SLOTS = 8
 /datum/preferences/New(client/C)
 	client=C
 	if(istype(C))
+		var/theckey = C.ckey
 		spawn()
 			while(!speciesinit)
 				sleep(1)
-			if(!IsGuestKey(C.key))
-				var/load_pref = load_preferences_sqlite(C.ckey)
+			if(!IsGuestKey(theckey))
```
Comment (gbasood) [inline code/modules/client/preferences.dm:190]: not sure why THIS line change is important

Comment (clusterfack) [inline code/modules/client/preferences.dm]: Because client can be null if someone logs out

**Comment by gbasood** [comment]
gotta merge because people can't ready up due to this bug

**Comment by Intigracy** [comment]
did it work

**Comment by Intigracy** [comment]
NAh it's still broken

**Pull request merged**